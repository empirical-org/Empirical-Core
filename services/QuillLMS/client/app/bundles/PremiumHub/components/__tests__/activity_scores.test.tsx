import * as React from "react";
import { render, } from "@testing-library/react";

import ActivityScores from '../activity_scores'

jest.mock('../../../Teacher/components/modules/user_is_premium', () => jest.fn());

const mockProps = {
  csvData: [],
  schoolNames: [],
  switchSchool: () => jest.fn,
  selectedSchool: '',
  teacherNames: [],
  switchTeacher: () => jest.fn,
  selectedTeacher: '',
  classroomNames: [],
  switchClassroom: () => jest.fn,
  selectedClassroom: '',
  filteredClassroomsData: [],
}

describe('ActivityScores', () => {
  test('it should render', () => {
    const { asFragment } = render(<ActivityScores {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
