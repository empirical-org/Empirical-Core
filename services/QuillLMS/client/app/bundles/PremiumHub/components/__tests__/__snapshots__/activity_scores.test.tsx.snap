// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityScores it should render 1`] = `
<DocumentFragment>
  <div
    class="admin-report-container"
  >
    <div
      class="header-container "
    >
      <div
        class="header-and-tooltip"
      >
        <h1>
          Activity Scores Report
        </h1>
        <span
          aria-expanded="false"
          aria-haspopup="true"
          class="quill-tooltip-trigger "
          role="button"
          tabindex="0"
        >
          <span>
            <img
              alt="Question mark icon"
              src="undefined/images/icons/icons-help.svg"
            />
          </span>
          <span
            class="quill-tooltip-wrapper"
          >
            <span
              aria-live="polite"
              class="quill-tooltip"
              id="tooltip"
              role="tooltip"
            />
          </span>
        </span>
      </div>
      <div
        class="csv-and-how-we-grade"
      >
        <button
          class="quill-button-archived focus-on-light small primary contained"
          style="display: block;"
        >
          Download report
        </button>
        <a
          class="how-we-grade focus-on-light"
          href="https://support.quill.org/activities-implementation/how-does-grading-work"
        >
          How we grade
        </a>
      </div>
    </div>
    <div
      class="dropdowns-container"
    >
      <div
        class="dropdown-container input-container inactive  not-editable admin-activity-dropdown bordered-dropdown  "
        role="button"
        tabindex="0"
      >
        <label />
        <div
          class="dropdown css-b62m3t-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-2-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="dropdown__control css-13cymwt-control"
          >
            <div
              class="dropdown__value-container css-1fdsijx-ValueContainer"
            >
              <div
                class="dropdown__placeholder css-1jqq78o-placeholder"
                id="react-select-2-placeholder"
              />
              <div
                class="dropdown__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-2-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="dropdown__input"
                  id="react-select-2-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="-1"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="dropdown__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="dropdown__indicator-separator css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="dropdown__indicator dropdown__dropdown-indicator css-1xc3v61-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="dropdown-container input-container inactive  not-editable admin-activity-dropdown bordered-dropdown  "
        role="button"
        tabindex="0"
      >
        <label />
        <div
          class="dropdown css-b62m3t-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-3-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="dropdown__control css-13cymwt-control"
          >
            <div
              class="dropdown__value-container css-1fdsijx-ValueContainer"
            >
              <div
                class="dropdown__placeholder css-1jqq78o-placeholder"
                id="react-select-3-placeholder"
              />
              <div
                class="dropdown__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-3-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="dropdown__input"
                  id="react-select-3-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="-1"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="dropdown__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="dropdown__indicator-separator css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="dropdown__indicator dropdown__dropdown-indicator css-1xc3v61-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="dropdown-container input-container inactive  not-editable admin-activity-dropdown bordered-dropdown  "
        role="button"
        tabindex="0"
      >
        <label />
        <div
          class="dropdown css-b62m3t-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-4-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="dropdown__control css-13cymwt-control"
          >
            <div
              class="dropdown__value-container css-1fdsijx-ValueContainer"
            >
              <div
                class="dropdown__placeholder css-1jqq78o-placeholder"
                id="react-select-4-placeholder"
              />
              <div
                class="dropdown__input-container css-qbdosj-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-4-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="dropdown__input"
                  id="react-select-4-input"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="-1"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="dropdown__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="dropdown__indicator-separator css-1u9des2-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="dropdown__indicator dropdown__dropdown-indicator css-1xc3v61-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="empty-state-for-report"
    >
      <img
        alt=""
        class="no-report-image"
        src="undefined/images/illustrations/empty-state-premium-reports.svg"
      />
      <h2>
        You have no reports yet!
      </h2>
      <p
        class="text-center"
      >
        Once your students have completed their assigned activities, their reports will be displayed here.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
